compass/ocean/tests/baroclinic_gyre/forward.py

```diff
@@ -0,0 +1,118 @@
+                      '20km': {'ntasks': 384,
+                               'min_tasks': 32,
+                               'dt': "'00:20:00'",
+                               'btr_dt': "'0000_00:00:20'",
+                               'mom_del4': "2.0e10 ",
+                               'run_duration': "'0000_00:01:00'"}}
+
+        if res_name not in res_params:
+            raise ValueError(
+                f'Unsupported resolution {res_name}. Supported values are: '
+                f'{list(res_params)}')
+
+        res_params = res_params[res_name]
+
+        ntasks = res_params['ntasks']
+        min_tasks = res_params['min_tasks']
+
+        super().__init__(test_case=test_case, name=name, subdir=subdir,
+                         ntasks=ntasks, min_tasks=min_tasks, openmp_threads=1)
+
+        # make sure output is double precision
+        self.add_streams_file('compass.ocean.streams', 'streams.output')
+
+        self.add_namelist_file('compass.ocean.tests.baroclinic_gyre',
+                               'namelist.forward')
+        if long:
+            output_interval = "0000-01-00_00:00:00"
+            restart_interval = "0001-00-00_00:00:00"
+        else:
+            output_interval = res_params['run_duration'].replace("'", "")
+            restart_interval = "0030_00:00:00"
+        replacements = dict(
+            output_interval=output_interval, restart_interval=restart_interval)
+        self.add_streams_file(package='compass.ocean.tests.baroclinic_gyre',
+                              streams='streams.forward',
+                              template_replacements=replacements)
+        options = dict()
+        for option in ['dt', 'btr_dt', 'mom_del4', 'run_duration']:
+            options[f'config_{option}'] = res_params[option]
+        if long:
+            # run for 3 years instead of 3 time steps
+            options['config_start_time'] = "'0001-01-01_00:00:00'"
+            options['config_stop_time'] = "'0004-01-01_00:00:00'"
+            options['config_run_duration'] = "'none'"
+
+        self.add_namelist_options(options=options)
+        self.add_input_file(filename='init.nc',
+                            target='../initial_state/initial_state.nc')
+        self.add_input_file(filename='forcing.nc',
+                            target='../initial_state/forcing.nc')
+        self.add_input_file(filename='graph.info',
+                            target='../initial_state/culled_graph.info')
+
+        self.add_model_as_input()
+
+        self.add_output_file(filename='output/output.0001-01-01_00.00.00.nc')
+
+    # no setup() is needed
+
+    def run(self):
+        """
+        Run this step of the test case
+        """
+        run_model(self, partition_graph=True)
+
+        if self.name == '3_year_test':
+            replacements = {'config_do_restart': '.true.',
+                            'config_start_time': "'file'"}
+            self.update_namelist_at_runtime(replacements)
```

Comment:
This isn't working because the name of the step is always `forward`, not `3_year_test`.  One fix is:
```suggestion
        if self.test_case.name == '3_year_test':
            replacements = {'config_do_restart': '.true.',
                            'config_start_time': "'file'"}
            self.update_namelist_at_runtime(replacements)
```
Or I think it might be more logical to store the `long` argument as an attribute and use it here:
```suggestion
        if self.long:
            replacements = {'config_do_restart': '.true.',
                            'config_start_time': "'file'"}
            self.update_namelist_at_runtime(replacements)
```